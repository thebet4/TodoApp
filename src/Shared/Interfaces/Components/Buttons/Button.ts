import { ElementType } from 'react';

export default interface IButtonProps {
  onPress?: () => void,
  Label?: string,
  CustomLeftIcon?: ElementType | ElementType<any> ;
  CustomRightIcon?: ElementType | ElementType<any> ;
};
