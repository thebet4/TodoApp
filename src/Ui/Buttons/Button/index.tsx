import React from 'react';
import IButtonProps from '../../../Shared/Interfaces/Components/Buttons/Button';
import { ButtonContainer, ButtonText } from './style';

const Button = ({
  onPress, Label, CustomLeftIcon, CustomRightIcon,
}: IButtonProps) => {
  return (
    <ButtonContainer onPress={onPress}>
      {
         CustomLeftIcon && <CustomLeftIcon style={{ marginRight: 12 }} width={24} height={24} />
        }
      <ButtonText>
        {Label}
      </ButtonText>
      {
         CustomRightIcon && <CustomRightIcon style={{ marginLeft: 12 }} width={24} height={24} />
        }
    </ButtonContainer>
  );
};

export default Button;
