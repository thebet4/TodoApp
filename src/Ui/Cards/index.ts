export { default as HotelCard } from './HotelCard';
export { default as TeamCard } from './TeamCard';
