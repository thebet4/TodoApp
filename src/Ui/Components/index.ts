export { default as Header } from './Header'
export { default as TabBarItem } from './TabBarItem'
