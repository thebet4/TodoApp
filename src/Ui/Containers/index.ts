export { default as Container } from './Container';
export { ContentContainer } from './ContentContainer';
