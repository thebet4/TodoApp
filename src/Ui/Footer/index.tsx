import React from 'react';
import {
  ImageBackground, View, TouchableOpacity, Image,
} from 'react-native';
import { Colors } from '../Theme';
import Caption from '../Typography/Caption';
import Title40 from '../Typography/Title40';

const Footer = () => {
  return (

    <ImageBackground
      source={require('./Assets/HomeFooterBg.png')}
      style={{ paddingBottom: 64 }}
    >
      <View style={{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 92,
      }}
      >
        <View style={{
          width: '30%',
          backgroundColor: Colors.Base.Light,
          height: 1,
        }}
        />
        <Title40 style={{
          color: Colors.Base.Light,
          marginHorizontal: 12,
        }}
        >
          Partners
        </Title40>
        <View style={{
          width: '30%',
          backgroundColor: Colors.Base.Light,
          height: 1,
        }}
        />
      </View>

      <View style={{
        alignItems: 'center',
        justifyContent: 'space-around',
      }}
      >
        <Image source={require('./Assets/partner1.png')} resizeMode="center" />
        <Image source={require('./Assets/partner2.png')} resizeMode="center" />
        <Image source={require('./Assets/partner4.png')} resizeMode="center" />
        <Image source={require('./Assets/partner5.png')} resizeMode="center" />
        <Image source={require('./Assets/partner6.png')} resizeMode="center" />
        <Image source={require('./Assets/partner7.png')} resizeMode="center" />

      </View>

      <View style={{ width: '70%', alignSelf: 'center' }}>

        <TouchableOpacity style={{
          height: 44,
          alignItems: 'center',
          justifyContent: 'center',
          borderWidth: 2,
          borderColor: Colors.Base.Light,
          borderRadius: 51,
          marginBottom: 17,
        }}
        >
          <Caption style={{ color: Colors.Base.Light, fontSize: 18 }}>
            Acessar
          </Caption>
        </TouchableOpacity>
        <TouchableOpacity style={{
          height: 44,
          alignItems: 'center',
          justifyContent: 'center',
          borderRadius: 51,
          backgroundColor: Colors.Base.Light,
        }}
        >
          <Caption style={{ color: Colors.Primary.Normal, fontSize: 18 }}>
            Abrir Conta
          </Caption>
        </TouchableOpacity>
      </View>
    </ImageBackground>

  );
};

export default Footer;
