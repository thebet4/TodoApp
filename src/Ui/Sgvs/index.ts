export { default as CalendarIcon } from './CalendarIcon'
export { default as ListIcon } from './ListIcon'
export { default as SunIcon } from './SunIcon'
export { default as TaskIcon } from './TaskIcon'
export { default as SearchIcon } from './SearchIcon'
export { default as FilterIcon } from './FilterIcon'
export { default as AddIcon } from './AddIcon'
export { default as PersonalIcon } from './PersonalIcon'
export { default as PriorityIcon } from './PriorityIcon'
export { default as ClockIcon } from './ClockIcon'
export { default as CommentIcon } from './CommentIcon'
export { default as RepeatIcon } from './RepeatIcon'
