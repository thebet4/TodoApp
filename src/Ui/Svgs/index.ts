export { default as LogoIcon } from './LogoIcon';
export { default as OpenedEyeIcon } from './OpenedEyeIcon';
export { default as CalendarIcon } from './CalendarIcon';
export { default as SearchIcon } from './SearchIcon';
export { default as NFTIcon } from './NFTIcon';
export { default as HotelIcon } from './HotelIcon';
export { default as BlockChainIcon } from './BlockChainIcon';
export { default as LogoWithTextIcon } from './LogoWithTextIcon';
