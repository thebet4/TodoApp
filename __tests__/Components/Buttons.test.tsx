import renderer from 'react-test-renderer';
import React from 'react';

import { Button } from '../../src/Ui/Buttons';

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
