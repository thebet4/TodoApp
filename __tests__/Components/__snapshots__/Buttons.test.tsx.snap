// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6A0DFB",
      "borderRadius": 100,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "marginBottom": 12,
      "marginTop": 12,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 21.7,
        },
      ]
    }
  />
</View>
`;
